### PR TITLE
Initial RPi Garage Door Commit

### PR DESCRIPTION
**Pull request by jslay88**
Some issues, please see issue #35

### CODE REVIEW

File: pyhap/accessories/GarageDoor.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
+import time
+import threading
+import pyhap.loader as loader
+import RPi.GPIO as GPIO
```
Comment (schinckel) [inline pyhap/accessories/GarageDoor.py]: If you use `gpiozero`, then it's possible to "swap out" the pin factory - this makes testing from another machine (where you have `pigpio` remote pins).

Comment (jslay88) [reply pyhap/accessories/GarageDoor.py]: Will look into it tomorrow or next. Need to rest  :)

Comment (schinckel) [reply pyhap/accessories/GarageDoor.py]: https://bitbucket.org/schinckel/garagedoor/src/e3268a75ca576f7dc65532adf1f6d51a8bc149fc/garage_door/accessory.py?at=default&fileviewer=file-view-default

Comment (schinckel) [reply pyhap/accessories/GarageDoor.py]: Oh, the other great thing about `gpiozero` is that you can set event handlers on pins.

In other words: "when the top trigger opens, this means the door is now closing", and so on.

Comment (jslay88) [reply pyhap/accessories/GarageDoor.py]: Yeah, thats basically what the callbacks are doing using the GPIO library. I only have one switch on my door, so I infer a lot here, on state and transition. lol.

Comment (jslay88) [reply pyhap/accessories/GarageDoor.py]: My excuse to learn a lot of different frameworks.

https://gitlab.com/jslay/garage_hub

Comment (schinckel) [reply pyhap/accessories/GarageDoor.py]: I'm happy to have one that works (a) with no switches: time delay based only, (b) a single switch, and (c) switches at both the top and bottom.

Comment (jslay88) [reply pyhap/accessories/GarageDoor.py]: yeah, thats one thing i was thinking about, how best to make this Accessory dynamic. 

I just started getting confused about what was happening with my issues listed above, that I figured I'd go ahead and post what I had so far.

**Comment by ikalchev** [comment]
Is this more as a need-info pull request?

**Comment by jslay88** [comment]
Yes, would possibly like to merge with master once refined.

**Comment by jslay88** [comment]
Not sure. Feel free to take what he has if you’d like.

I mainly needed a software debounce method as all of the GPIO libraries
that do support debounce for Pi, have the same issue with latent functions
as the callback.

Then was having weird issues and needed some help since I don’t have
documentation on how these are expected to work.

**Comment by jslay88** [comment]
I have no problem adding onto what he has done to get what I need out of
it. I think we were also beginning to discuss what we want in the
GarageDoor Accessory. So we are likely to build off of whatever gets
merged. We just want to provide you with what we are working on, just funny
we happen to literally did the 2 same things with no background each other.
Lol

**Comment by ikalchev** [comment]
Haha okay. Well, you both are driving this, so it would be perfect if you can say how you prefer to do the merges :)

I see that #42 is at a very finished state with docs and stuff, but it is not for GPIO specifically. Perhaps we can again split the work in two changes, i.e. 42 and your DoorSwitch here (Correct me if I am misunderstanding the commits here, I haven't had the chance to get into it more deeply).

As I said, I have no preference, so if you two have a plan on how to merge these or need help, just let me know :)

**Comment by jslay88** [comment]
Yeah for sure his is in a more complete state. I’d say take his for the
ground work and I can work with him to add in what I need out of it.

**Comment by ikalchev** [comment]
Sounds good. There is just one question from my side at the review, so its pretty much there.

**Pull request closed without merging**